datafusion/src/logical_plan/expr.rs

```diff
@@ -1408,6 +1452,11 @@ pub fn lit<T: Literal>(n: T) -> Expr {
     n.lit()
 }
 
+/// Create a literal timestamp expression
+pub fn lit_timestamp_nanosecond<T: Literal>(n: T) -> Expr {
```

Comment:
do we consider it a user error if a non-numerical literal type is passed in here? If so, it might be better to introduce a new ns timestamp trait and only implement it for the numeric types we support so we can let rust compiler catch these issues at build time.

Reply:
I think it would be nice to try and parse `str` into timestamps, though that would require making this function fallible, which might be kind of messy. 🤔

Reply:
I think a new trait for appropriate numeric only is a good solution. It can be extended later to support string parsing. Let me give it a shot.